### ISSUE TITLE
Unchecking 'User-Unhandled Exceptions' doesn't work

### ISSUE THREAD

**Issue by mdowais** [OmniSharp/omnisharp-vscode#3388]
## Issue Description ##
The breakpoint panel in the debug sidebar has an option which allows disabling  breakpoint on exception. But this doesnt seem to work. I tried to disable User Unhandled exception so that it doesn't break at this type of exception and let the webserver handle the exception (ASP.NET CORE Application)

## Steps to Reproduce ##
throw an exception through code
## Expected Behavior ##
When the tick box is unchecked, it should not throw the exception
## Actual Behavior ##
When the tick box is unchecked, it is throwing the exception
### OmniSharp log ###
<details>Post the output from Output-->OmniSharp log here</details>

### C# log ###
<details>Post the output from Output-->C# here</details>

## Environment information ##

**VSCode version**: 1.39.2
**C# Extension**: 1.21.5

<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100
 Commit:    04339c3a26

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100\

Host (useful for support):
  Version: 3.0.0
  Commit:  7d57652f33

.NET Core SDKs installed:
  2.1.700 [C:\Program Files\dotnet\sdk]
  2.2.107 [C:\Program Files\dotnet\sdk]
  3.0.100 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|Angular-BeastCode|Mikael|8.0.10|
|angular-console|nrwl|8.1.2|
|angular-essentials|johnpapa|0.6.3|
|Angular2|johnpapa|8.1.1|
|angular2-inline|natewallace|0.0.17|
|angular2-switcher|infinity1207|0.2.0|
|awesome-flutter-snippets|Nash|2.0.3|
|better-comments|aaron-bond|2.0.5|
|bracket-pair-colorizer-2|CoenraadS|0.0.28|
|cpptools|ms-vscode|0.26.0|
|csharp|ms-vscode|1.21.5|
|csharpfixformat|Leopotam|0.0.84|
|dart-code|Dart-Code|3.5.1|
|debugger-for-chrome|msjsdiag|4.12.0|
|EditorConfig|EditorConfig|0.14.2|
|flutter|Dart-Code|3.5.1|
|git-graph|mhutchie|1.17.0|
|gitlens|eamodio|10.1.1|
|java|redhat|0.51.0|
|material-icon-theme|PKief|3.9.1|
|metago|metaseed|2.11.0|
|msbuild-project-tools|tintoy|0.2.55|
|mssql|ms-mssql|1.7.0|
|netcore-snippets|doggy8088|0.2.7|
|ng-template|Angular|0.802.3|
|ng-zorro-vscode|cipchk|8.3.1|
|polacode|pnp|0.3.4|
|prettier-vscode|esbenp|2.3.0|
|pubspec-assist|jeroen-meijer|0.3.6|
|pubspec-dependency-search|everettjf|1.0.0|
|python|ms-python|2019.10.41019|
|quantum-devkit-vscode|quantum|0.9.19093002|
|quicktype|quicktype|12.0.46|
|rest-client|humao|0.22.2|
|search-node-modules|jasonnutter|1.3.0|
|sqltools|mtxr|0.21.0|
|svn-scm|johnstoncode|1.54.6|
|todo-tree|Gruntfuggly|0.0.160|
|vscode-java-debug|vscjava|0.22.0|
|vscode-java-dependency|vscjava|0.6.0|
|vscode-java-pack|vscjava|0.8.0|
|vscode-maven|vscjava|0.19.1|
|vscode-npm-script|eg2|0.3.9|
|vscode-nuget-package-manager|jmrog|1.1.6|
|vscode-scss|mrmlnc|0.6.2|
|vscode-typescript-tslint-plugin|ms-vscode|1.2.2|
|vscode-versionlens|pflannery|0.24.0|
|vscodeintellicode|VisualStudioExptTeam|1.2.0|
|xml|DotJoshJohnson|2.5.0|;

</details>

**Comment by gregg-miskelly** [comment]
The debugger will always stop on exceptions that will crash your application. You wouldn't want this to be disabled - your app would just exit.

If this isn't what you are asking about, please provide real repro steps.

**Comment by mdowais** [comment]
This is an ASP.Net Core application, say the application has thrown an unhandled exception which wont make the app crash because it is running on the kestral web server, if i manually click on continue the browser shows the error page with stack trace and details.

My question is, that if there is a check box which says break on unhandled exception, when i untick this, why would it still break at an unhandled exception?

**Comment by mdowais** [comment]
And a friend of mine was playing around with it, and found that turning the checkbox on and off for a few time, made it work for him.

@vscodeguru

**Comment by gregg-miskelly** [comment]
Thanks. I see what you mean. I am not sure how this used to work, but it looks like the issue is that we get an empty collection of 'filters' from VS Code and our code doesn't seem to handle that correctly.

```
-> (C) {"command":"setExceptionBreakpoints","arguments":{"filters":[]},"type":"request","seq":18}
<- (R) {"seq":342,"type":"response","request_seq":18,"success":true,"command":"setExceptionBreakpoints"}
```

**Comment by gregg-miskelly** [comment]
In case this is useful: If you uncheck 'User-Unhandled' before you start debugging you can work around the bug. I realize this isn't a great work around, but hopefully we will soon put out a fix.

**Comment by mdowais** [comment]
> In case this is useful: If you uncheck 'User-Unhandled' before you start debugging you can work around the bug. I realize this isn't a great work around, but hopefully we will soon put out a fix.

Nope, this doesnt seem to work, 

Turned off User Unhanled Exception before debugging, but still i get a break.
![Untitled](https://user-images.githubusercontent.com/4958982/68067694-1bbcca80-fd71-11e9-9355-2b6e05345574.png)

**Comment by gregg-miskelly** [comment]
Fixed in 1.21.
